validation/charts/appco/istio_test.go

```diff
@@ -0,0 +1,220 @@
+	err = createIstioNamespace(client, i.cluster.ID)
+	require.NoError(i.T(), err)
+
+	i.T().Logf("Creating %s secret", RancherIstioSecret)
+	logCmd, err := createIstioSecret(client, i.cluster.ID, *AppCoUsername, *AppCoAccessToken)
+	require.NoError(i.T(), err)
+	require.True(i.T(), strings.Contains(logCmd, RancherIstioSecret))
+
+	ambientSets := `--set cni.enabled=true,ztunnel.enabled=true --set istiod.cni.enabled=false --set cni.profile=ambient,istiod.profile=ambient,ztunnel.profile=ambient`
+
+	i.T().Log("Installing Ambient Istio AppCo")
+	istioChart, logCmd, err := installIstioAppCo(client, i.cluster.ID, *AppCoUsername, *AppCoAccessToken, ambientSets)
+	require.NoError(i.T(), err)
+	require.True(i.T(), strings.Contains(logCmd, "deployed"))
+	require.True(i.T(), istioChart.IsAlreadyInstalled)
+}
+
+func (i *IstioTestSuite) TestGatewayStandaloneInstallation() {
+	subSession := i.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := i.client.WithSession(subSession)
+	require.NoError(i.T(), err)
+
+	i.T().Logf("Creating %s namespace", charts.RancherIstioNamespace)
+	err = createIstioNamespace(client, i.cluster.ID)
+	require.NoError(i.T(), err)
+
+	i.T().Logf("Creating %s secret", RancherIstioSecret)
+	logCmd, err := createIstioSecret(client, i.cluster.ID, *AppCoUsername, *AppCoAccessToken)
+	require.NoError(i.T(), err)
+	require.True(i.T(), strings.Contains(logCmd, RancherIstioSecret))
+
+	gatewaySets := `--set base.enabled=false,istiod.enabled=false --set gateway.enabled=true,gateway.namespaceOverride=default`
+
+	i.T().Log("Installing Gateway Standalone Istio AppCo")
+	istioChart, logCmd, err := installIstioAppCo(client, i.cluster.ID, *AppCoUsername, *AppCoAccessToken, gatewaySets)
+	require.NoError(i.T(), err)
+	require.True(i.T(), strings.Contains(logCmd, expectedDeployLog))
+	require.True(i.T(), istioChart.IsAlreadyInstalled)
+}
+
+func (i *IstioTestSuite) TestGatewayDiffNamespaceInstallation() {
+	subSession := i.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := i.client.WithSession(subSession)
+	require.NoError(i.T(), err)
+
+	i.T().Logf("Creating %s namespace", charts.RancherIstioNamespace)
+	err = createIstioNamespace(client, i.cluster.ID)
+	require.NoError(i.T(), err)
+
+	i.T().Logf("Creating %s secret", RancherIstioSecret)
+	logCmd, err := createIstioSecret(client, i.cluster.ID, *AppCoUsername, *AppCoAccessToken)
+	require.NoError(i.T(), err)
+	require.True(i.T(), strings.Contains(logCmd, RancherIstioSecret))
+
+	gatewayNamespaceSets := `--set gateway.enabled=true,gateway.namespaceOverride=default`
+
+	i.T().Log("Installing Gateway Namespace Istio AppCo")
+	istioChart, logCmd, err := installIstioAppCo(client, i.cluster.ID, *AppCoUsername, *AppCoAccessToken, gatewayNamespaceSets)
+	require.NoError(i.T(), err)
+	require.True(i.T(), strings.Contains(logCmd, expectedDeployLog))
+	require.True(i.T(), istioChart.IsAlreadyInstalled)
+}
+
+func (i *IstioTestSuite) TestCanaryUpgrade() {
```

Comment:
this appears to be `installIstioUpgradeWithCanary` or something similar, so that the name can reflect that
1. the chart will have a clean install as part of the test
2. upgrade will install / configure canary after the install is successful.